integration/mediation-tests/tests-platform/tests-transaction/src/test/java/org/wso2/micro/integrator/transaction/TransactionCounterUtils.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.transaction;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.awaitility.Awaitility;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Transaction Counter test Utils
+ */
+public class TransactionCounterUtils {
+    private final static JsonParser jsonParser = new JsonParser();
+    private final static int POLL_INTERVAL = 2;
+    private final static int POLL_TIMEOUT = 3;
+
+    public static void awaitForTransactionCountUpdate(String transactionCountResource, int requestCount) {
+        Awaitility.await().
+                pollInterval( POLL_INTERVAL, TimeUnit.SECONDS).
+                atMost(POLL_TIMEOUT, TimeUnit.MINUTES).
+                until(assertTransactionCount(transactionCountResource, requestCount));
+    }
+
+    private static Callable<Boolean> assertTransactionCount(String transactionCountResource, int expectedTransactionCount) {
+        return new Callable<Boolean>() {
+            @Override
+            public Boolean call() throws Exception {
+                return expectedTransactionCount == getTransactionCount(transactionCountResource);
```

Comment:
when this fails , can we print the no. of transactions available in db

Reply:
Fixed